src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = '0.5',
+  slippageMode = 'Auto',
+}: SwapSettingsSlippageInputReact) {
+  const [customSlippage, setCustomSlippage] = useState(defaultSlippage);
+
+  return (
+    <div
+      className={cn(
+        'flex items-center justify-between rounded-lg border border-gray-300 bg-white px-2 py-1 dark:border-gray-700 dark:bg-gray-950',
+        className,
+      )}
+    >
+      <input
+        type="text"
+        value={customSlippage}
+        onChange={(e) => setCustomSlippage(e.target.value)}
+        className="w-12 bg-transparent pl-1 font-normal font-sans text-gray-900 text-sm leading-6 focus:outline-none dark:text-gray-50"
+        disabled={slippageMode === 'Auto'}
```

Comment:
This all `Auto` part smells a bit... like what happen if I write in `slippageMode` the word `potato`.

Feels an odd developer experience.